Sources_v3/StreamChatUI/ChatChannel/ChatChannelVC.swift

```diff
@@ -241,6 +232,100 @@ open class ChatChannelVC<ExtraData: UIExtraDataTypes>: ViewController,
             isLastInGroup: isLastInGroup
         )
     }
+
+    private func actionsController(
+        for message: _ChatMessageGroupPart<ExtraData>,
+        messageController: _ChatMessageController<ExtraData>
+    ) -> ChatMessageActionsViewController<ExtraData> {
+        var messageActions: [ChatMessageActionItem] {
+            guard
+                let message = messageController.message,
+                let currentUser = messageController.client.currentUserController().currentUser
+            else { return [] }
+
+            var actions: [ChatMessageActionItem] = []
+
+            actions.append(.inlineReply { [weak self] in
+                debugPrint("inline reply")
+                self?.dismiss(animated: true)
+            })
+
+            actions.append(.threadReply { [weak self] in
+                debugPrint("thread reply")
+                self?.dismiss(animated: true)
+            })
+
+            actions.append(.copy { [weak self] in
+                UIPasteboard.general.string = message.text
+                self?.dismiss(animated: true)
+            })
+
+            if message.isSentByCurrentUser {
+                actions.append(.edit { [weak self] in
+                    debugPrint("edit")
+                    self?.dismiss(animated: true)
+                })
+                actions.append(.delete { [weak self] in
+                    debugPrint("delete")
+                    messageController.deleteMessage { [messageController] _ in
+                        self?.dismiss(animated: true)
+                        _ = messageController
+                    }
+                })
+            } else {
+                if currentUser.mutedUsers.contains(message.author) {
+                    actions.append(.unmuteUser { [weak self] in
+                        let userController = messageController.client.userController(userId: message.author.id)
+                        userController.unmute { [userController] _ in
+                            self?.dismiss(animated: true)
+                            _ = userController
+                        }
+                    })
+                } else {
+                    actions.append(.muteUser { [weak self] in
+                        let userController = messageController.client.userController(userId: message.author.id)
+                        userController.mute { [userController] _ in
+                            self?.dismiss(animated: true)
+                            _ = userController
+                        }
+                    })
+                }
+            }
+
+            return actions
+        }
+
+        return .init(
+            view: uiConfig.messageList.messageActionsView.init().withoutAutoresizingMaskConstraints,
```

Comment:
Why are you injecting view from outside?

Reply:
`ChatMessageActionsViewController` doesn't inherit from `UIViewController` and doesn't serve as a `UIConfigProvider` so we cannot instantiate view of correct type without passing the `UIConfig` there. 

Decided to keep things simple and inject the views.

Reply:
-> Slack